server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -170,10 +162,17 @@ public SearchHit(StreamInput in) throws IOException {
             documentFields = in.readMap(StreamInput::readString, DocumentField::new);
             metaFields = in.readMap(StreamInput::readString, DocumentField::new);
         } else {
+            // TODO: remove this part in v 9.0
             Map<String, DocumentField> fields = readFields(in);
-            documentFields = new HashMap<>();
-            metaFields = new HashMap<>();
-            SearchHit.splitFieldsByMetadata(fields, documentFields, metaFields);
+            for (DocumentField docField : fields.values()) {
+                if (docField.isMetadataField()) {
```

Comment:
Is this the (or at least one) spot where we need the metadata flag on the DocumentField? Looks like from v7.8 on we can already read into separate maps, so this would only be needed before, and it looks like in that case DocumentField(StreamInput in) always uses  MapperService.isMetadataFieldSt to determine the fields meta-status?